common/org.testeditor.dsl.common.ui/src/org/testeditor/dsl/common/ui/wizards/NewProjectWizard.xtend

```diff
@@ -36,45 +30,41 @@ import org.testeditor.dsl.common.ui.utils.ProjectContentGenerator
  */
 class NewProjectWizard extends BasicNewProjectResourceWizard {
 
-	@Inject extension ProjectUtils
 	TestProjectConfigurationWizardPage configPage
 
 	@Inject ProjectContentGenerator projectContentGenerator
 	@Inject ProgressMonitorRunner progressMonitorRunner
 
-	private def void addNature(IProject newProject, String nature) {
-		if (!newProject.hasNature(nature)) {
-			val description = newProject.getDescription
-			description.setNatureIds(description.getNatureIds + #[nature])
-			newProject.setDescription(description, null)
-		}
-	}
-
 	override init(IWorkbench bench, IStructuredSelection selection) {
 		super.init(bench, selection)
 		windowTitle = "Create test-first project"
 	}
 
 	override addPages() {
-		super.addPages()
-		configPage = new TestProjectConfigurationWizardPage("configPage")
-		configPage.availableBuildSystems = projectContentGenerator.availableBuildSystems
-		configPage.availableFixtureNames = projectContentGenerator.availableFixtureNames
-		addPage(configPage)
+		super.addPages
+		configPage = new TestProjectConfigurationWizardPage("configPage") => [
+			availableBuildSystems = projectContentGenerator.availableBuildSystems
+			availableFixtureNames = projectContentGenerator.availableFixtureNames
+			addPage
+		]
 	}
 
 	override getNextPage(IWizardPage page) {
 		return configPage
 	}
+	
+	override canFinish(){
+		val projectName=(pages.get(0) as WizardNewProjectCreationPage).projectName
+		val nameOk=projectName.matches("^[a-zA-Z_0-9]+$")
+		if(!nameOk && !projectName.empty){
```

Comment:
The second check is good to have no error message when entering the wizard (with an empty project name). The finish button will not be active in case of an empty project name.